ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+            if self.exitThread:
+                break
+        logger.info("Logger thread done printing flexy logs")
+        # exit only this thread
+        sys.exit()
+
+    def build_container_args(self):
+        """
+        Builds most commonly used arguments for flexy container
+
+        """
+        args = list()
+        args.append(f"--log-opt={self.flexy_log_file}")
+        args.append(f"--env-file={self.flex_env_file}")
+        args.append(f"-w={self.flexy_mnt_container_dir}")
+        args.append(
+            f"--mount=type=bind,source={self.flexy_mnt_host_dir},"
+            f"destination=f{self.flexy_mnt_container_dir},relabel=shared"
+        )
+        args.append(
+            f"--mount=type=bind,source={self.flexy_host_private_conf_dir_path},"
+            f"destination={self.flexy_mnt_private_conf_dir},relabel=shared"
+        )
+        args.append(f"{self.flexy_img_url}")
+        return args
+
+    def clone_and_unlock_ocs_private_conf(self):
+        """
+        Clone ocs_private_conf (flexy env and config) repo into
+        flexy_mnt_host_dir
+
+        """
+        clone_repo(self.flexy_private_conf_url, self.flexy_host_private_conf_dir_path)
+        # git-crypt unlock /path/to/keyfile
+        old_cwd = os.getcwd()
+        os.chdir(self.flexy_host_private_conf_dir_path)
+        cp = subprocess.run(
+            f'git-crypt unlock {constants.FLEXY_GIT_CRYPT_KEYFILE}',
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        if cp.returncode:
+            raise exceptions.CommandFailed(cp.stderr)
+        logger.info("Unlocked the git repo")
+        os.chdir(old_cwd)
+
+    def merge_flexy_env(self):
+        """
+        Update the ocs-osp.env file with the user supplied values.
+        This function assumes that flexy-ocs-private repo has been
+        already cloned
+
+        """
+        config_parser = configparser.ConfigParser()
+        # without this,config parser would convert all
+        # uppercase keys to lowercase
+        config_parser.optionxform = str
+        env_file = os.path.join(
+            self.flexy_host_private_conf_dir_path, constants.FLEXY_DEFAULT_ENV_FILE
+        )
+
+        with open(env_file, "r") as fp:
+            # we need to add section to make
+            # config parser happy, this will be removed
+            # while writing back to file
+            file_content = "[root]\n" + fp.read()
+
+        config_parser.read_string(file_content)
+        # Iterate over config_parser keys, if same key is present
+        # in user supplied dict update config_parser
+        for ele in config_parser.items('root'):
+            if ele[0] in config.ENV_DATA:
+                # For LAUNCHER_VARS we need to merge the
+                # user provided dict with default obtained
+                # from env file
+                if ele[0] == 'LAUNCHER_VARS':
+                    merge_dict(config.ENV_DATA[ele[0]], ele[1])
```

Comment:
If I understand it correctly, the configuration of `LAUNCHER_VARS` in *ocs-osp.env* have precedence here before the onfiguration in the `config.ENV_DATA`, right? In other words if I'll provide my own dict in `config.ENV_DATA['LAUNCHER_VARS']`, existing keys in the `LAUNCHER_VARS` variable in *ocs-osp.env* file will not be updated?
If the above statement is correct, wouldn't be better to make it vice versa - give higher priority to the configuration provided by user in the `config.ENV_DATA['LAUNCHER_VARS']`?

Reply:
You are right, I visualized it in wrong way, I will change that.

Reply:
Done, did it in the correct order now.